File: Nordstrom/examples/ex-named.js
#!/usr/bin/env node

const AWS = require('aws-sdk')

getCredentials = async () => {
  return new Promise(function (resolve, reject) {
    process.env.AWS_SDK_LOAD_CONFIG = true
    var creds = new AWS.ProcessCredentials({ profile: 'nordstrom-federated' })
    console.log(creds)
    AWS.config.credentials = creds
    AWS.config.region = 'us-west-2'
    AWS.config.getCredentials(function (error, response) {
      if (error) {
        reject(new Error(error.stack))
      }
      resolve(response)
    })
  })
}

listBuckets = async () => {
  return new Promise(function (resolve, reject) {
    var s3 = new AWS.S3()
    var bucketInfo = s3.listBuckets(function (error, response) {
      if (error) {
        reject(new Error(error.stack))
      }
      resolve(response)
    })
  })
}

(async () => {
  creds = await getCredentials()
  console.log(creds)
  var buckets = await listBuckets()
  console.log(buckets)
})().catch(error => {
  console.error(error)
})

File: Nordstrom/examples/ex-named.go
package main

import (
	"fmt"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/awserr"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3"
)

func main() {
	s, err := session.NewSessionWithOptions(
		session.Options{
			SharedConfigState: session.SharedConfigEnable,
			Profile:           "nordstrom-federated", //omit or "default" if using default profile
			Config: aws.Config{
				Region:                        aws.String("us-west-2"),
				CredentialsChainVerboseErrors: aws.Bool(true),
			},
		},
	)
	if err != nil {
		fmt.Println(err.Error())
	}

	svc := s3.New(s)

	resp, err := svc.ListBuckets(
		&s3.ListBucketsInput{},
	)
	if err != nil {
		if aerr, ok := err.(awserr.Error); ok {
			switch aerr.Code() {
			default:
				fmt.Println(aerr.Error())
			}
		} else {
			fmt.Println(err.Error())
		}
		return
	}
	fmt.Println(resp)
}
